File: (6 kyu) CamelCase Method.js
/** CamelCase Method (6 kyu) https://www.codewars.com/kata/camelcase-method/
 * Write simple .camelcase method (camel_case function in PHP) for strings. All
 * words must have their first letter capitalized without spaces.
 * 
 * Example:
 * "hello case".camelCase() => HelloCase
 * "camel case word".camelCase() => CamelCaseWord
**/
String.prototype.camelCase = function() {
  return this
          .split(' ')
          .map((str) => {
            return str
                    .charAt(0)
                    .toUpperCase() + str.slice(1);
          })
          .join('');
}

File: (6 kyu) CamelCase Method/(6 kyu) CamelCase Method.rs
fn camel_case(str: &str) -> String {
    str.split_whitespace()
        .collect::<Vec<_>>()
        .iter()
        .map(|&s| [&s[..1].to_uppercase(), &s[1..]].join(""))
        .collect::<Vec<_>>()
        .join("")
}
